src/background/modules/InstallUpgrade.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Upgrades user data on installation of new updates.
+ *
+ * @module InstallUpgrade
+ */
+
+import * as Notifications from "/common/modules/Notifications.js";
+import { getBrowserValue } from "/common/modules/BrowserCompat.js";
+
+
+const notifications = new Map();
+
+
+browser.notifications.onClicked.addListener((notificationId) => {
+    const url = notifications.get(notificationId);
+
+    if (url) {
+        browser.tabs.create({ url });
+    }
+});
+
+browser.notifications.onClosed.addListener((notificationId) => {
+    notifications.delete(notificationId);
+});
+
+/**
+ * Checks whether an upgrade is needed.
+ *
+ * @see {@link https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/runtime/onInstalled}
+ * @private
+ * @param {Object} details
+ * @returns {void}
+ */
+function handleInstalled(details) {
+    console.log(details);
+
+    const manifest = browser.runtime.getManifest();
+    switch (details.reason) {
+    case "install":
+        // TODO: This will need to be localized
+        Notifications.showNotification(`🎉 ${manifest.name} installed`, `Thank you for installing the “${manifest.name}” add-on!\nVersion: ${manifest.version}\n\nOpen the options/preferences page to configure this extension.`);
+        break;
+    case "update":
+        if (Notifications.SEND) {
```

Comment:
I don't really get what's the `SEND` variable for? Where is it modified and why cannot you just call a function?

It alos needs a better name if it is there to say, what does it send where/how/when? What does it even control?
I'd guess it goes into some `shouldSendNotification` direction or so,

Also again clran code-like I'd say either early return here (which may indeed be inconvenient here, if more upgrade code/logic is added, or make that whole send functionality or so into another extra function – or even file/"module", if needed.

Reply:
The `SEND` variable is from the Notifications module. Specifically, it is the value of the new option I added to disable desktop notifications (i.e. whether to "send" the notifications).

Anyway, this is the section I suspected you would want to refactor into the existing Notifications module as part of https://github.com/TinyWebEx/RandomTips/issues/9, but first we would of course need to agree on a new interface for that module. Ideas are welcome...